sdk/eventgrid/azure-eventgrid/tests/test_eg_publisher_client.py

```diff
@@ -0,0 +1,82 @@
+#-------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+#--------------------------------------------------------------------------
+
+import logging
+import sys
+import os
+import pytest
+
+from devtools_testutils import AzureMgmtTestCase, ResourceGroupPreparer
+
+from azure.core.credentials import AzureKeyCredential
+from azure.eventgrid import EventGridPublisherClient, CloudEvent, EventGridEvent
+
+from eventgrid_preparer import (
+    EventGridTopicPreparer
+)
+
+
+class EventGridPublisherClientTests(AzureMgmtTestCase):
+    @pytest.mark.liveTest
+    @ResourceGroupPreparer(name_prefix='eventgridtest')
```

Comment:
Can import and use the CachedResourceGroupPreparer (in addition to using the cached functionality within your EG Topic Preparer as well) to speed these up significantly.